plugins/snapshot/local_snapshot.go

```diff
@@ -116,20 +116,41 @@ func getMilestoneApprovees(milestoneIndex milestone_index.MilestoneIndex, cached
 			} else {
 				// Tx is not confirmed
 				// ToDo: This shouldn't happen, but it does since tipselection allows it at the moment
-				if !ignoreConfirmationState {
-					cachedTx.Release(true) // tx -1
-					return nil, errors.Wrapf(ErrUnconfirmedTxInSubtangle, ": %v", txHash)
-				}
-
-				// Check if the we can walk further
-				if !tangle.ContainsTransaction(cachedTx.GetTransaction().GetTrunk()) {
-					cachedTx.Release(true) // tx -1
-					continue
-				}
-
-				if !tangle.ContainsTransaction(cachedTx.GetTransaction().GetBranch()) {
+				if !collectForPruning {
+					if cachedTx.GetTransaction().IsTail() {
+						cachedTx.Release(true) // tx -1
+						log.Panicf("getMilestoneApprovees: Transaction not confirmed: %v", txHash)
+					}
+
+					// Search all references tails of this Tx (needed for correct SolidEntryPoint calculation)
+					tailTxs, err := dag.FindAllTails(txHash, true)
+					if err != nil {
+						cachedTx.Release(true) // tx -1
+						return nil, err
+					}
+
+					for tailTx := range tailTxs {
+						txsToTraverse[tailTx] = struct{}{}
+					}
+
+					// Ignore this transaction in the cone because it is not confirmed
 					cachedTx.Release(true) // tx -1
 					continue
+				} else {
```

Comment:
This else statement is redundant since the if always continues the loop or returns out of the function

Reply:
Good catch